.github/workflows/test-sui.yaml

```diff
@@ -16,7 +10,27 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Install Node.js
+      - name: Check for Relevant Changes
+        uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            sui:
+              - 'sui/**'
+            common:
+              - 'common/**'
+            github:
+              - '.github/actions/setup-sui/**'
+              - '.github/workflows/test-sui.yaml'
+
+      - name: Summarize Changes
+        run: |
+          echo "Changes in sui: ${{ steps.filter.outputs.sui }}"
+          echo "Changes in common: ${{ steps.filter.outputs.common }}"
+          echo "Changes in github: ${{ steps.filter.outputs.github }}"
+
+      - name: Setup Node.js
+        if: steps.filter.outputs.sui == 'true' || steps.filter.outputs.common == 'true' || steps.filter.outputs.github == 'true'
```

Comment:
will this if only skip this step or all the rest as well?

Reply:
actually, I concern about this after commit as well 😅. updated in the commit below.